### PR TITLE
Adding functionalities to datapar needed by octotiger

### PR DESCRIPTION
**Pull request by hkaiser**
- adding datapar serialization
- adding fixed size arrays support from Vc
- enable using datapar_execution on input ranges over vector packs (Vc only for now)

### CODE REVIEW

File: hpx/parallel/datapar/execution_policy.hpp [hunk 1]
```diff
@@ -26,17 +27,18 @@
 namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v1)
 {
     ///////////////////////////////////////////////////////////////////////////
-    /// Extension: The class datapar_task_execution_policy is an execution
+    /// Extension: The class sequential_task_execution_policy is an execution
```
Comment (sithhell) [inline hpx/parallel/datapar/execution_policy.hpp]: why the change from `datapar_task_execution_policy`? Shouldn't it be `dataseq_task_execution_policy`?

File: hpx/parallel/datapar/execution_policy.hpp [hunk 2]
```diff
@@ -122,17 +126,17 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v1)
...
         ///                 different parameter types can't be duplicated
         ///
-        /// \returns The new parallel_execution_policy_shim
+        /// \returns The new dataseq_task_execution_policy
         ///
         template <typename... Parameters, typename ParametersType =
             typename executor_parameters_join<Parameters...>::type>
         typename rebind_executor<
-            datapar_task_execution_policy, executor_type, ParametersType
+            sequential_task_execution_policy, executor_type, ParametersType
```
Comment (sithhell) [inline hpx/parallel/datapar/execution_policy.hpp]: shouldn't that be `dataseq_task_execution_policy`?

File: hpx/parallel/datapar/execution_policy.hpp [hunk 3]
```diff
@@ -122,17 +126,17 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v1)
...
         >::type
         with(Parameters &&... params) const
         {
             typedef typename rebind_executor<
-                datapar_task_execution_policy, executor_type, ParametersType
+                sequential_task_execution_policy, executor_type, ParametersType
```
Comment (sithhell) [inline hpx/parallel/datapar/execution_policy.hpp]: same as above

File: hpx/parallel/datapar/execution_policy.hpp [hunk 4]
```diff
@@ -289,44 +450,885 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v1)
...
+        {};
+        /// \endcond
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    namespace detail
+    {
+        /// \cond NOINTERNAL
+        template <>
+        struct is_datapar_execution_policy<dataseq_execution_policy>
```
Comment (sithhell) [inline hpx/parallel/datapar/execution_policy.hpp]: so dataseq is a datapar execution policy? That sounds confusing.

File: hpx/parallel/datapar/execution_policy.hpp (hunk 2 above)
Comment (hkaiser) [inline hpx/parallel/datapar/execution_policy.hpp]: Thanks, copy&paste error.

File: hpx/parallel/datapar/execution_policy.hpp (hunk 4 above)
Comment (hkaiser) [inline hpx/parallel/datapar/execution_policy.hpp]: `is_datapar_execution_policy` should probably be renamed to something else, I'm not sure what name to use, though. Any ideas?

File: hpx/parallel/datapar/execution_policy.hpp (hunk 1 above)
Comment (hkaiser) [inline hpx/parallel/datapar/execution_policy.hpp]: Stupid mistake, thanks

**Comment by hkaiser** [comment]
@sithhell all review comments have been addressed
